presto-orc/src/main/java/com/facebook/presto/orc/OrcReader.java

```diff
@@ -111,12 +111,18 @@ public OrcReader(OrcDataSource orcDataSource, MetadataReader delegate, DataSize
         // get length of PostScript - last byte of the file
         int postScriptSize = buffer[buffer.length - SIZE_OF_BYTE] & 0xff;
 
-        // make sure this is an ORC file and not an RCFile or something else
-        verifyOrcFooter(orcDataSource, postScriptSize, buffer);
-
         // decode the post script
-        int postScriptOffset = buffer.length - SIZE_OF_BYTE - postScriptSize;
-        PostScript postScript = metadataReader.readPostScript(buffer, postScriptOffset, postScriptSize);
+        PostScript postScript;
+        try {
+            postScript = metadataReader.readPostScript(buffer, buffer.length - SIZE_OF_BYTE - postScriptSize, postScriptSize);
```

Comment:
Is there a chance it is not a valid Orc file but this call doesn't fail?

Reply:
No.. it is protobuf with multiple fields